client/rest-high-level/src/main/java/org/elasticsearch/client/RestHighLevelClient.java

```diff
@@ -359,6 +361,27 @@ public final void deleteAsync(DeleteRequest deleteRequest, ActionListener<Delete
             Collections.singleton(404), headers);
     }
 
+    /**
+     * Opens an index using the Open Index API
+     *
+     * See <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-open-close.html">
+     * Open Index API on elastic.co</a>
+     */
+    public final OpenIndexResponse openIndex(OpenIndexRequest openIndexRequest, Header... headers) throws IOException {
+        return performRequestAndParseEntity(openIndexRequest, Request::openIndex, OpenIndexResponse::fromXContent, emptySet(), headers);
+    }
+
+    /**
+     * Asynchronously opens an index using the Open Index API
+     *
+     * See <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-open-close.html">
+     * Open Index API on elastic.co</a>
+     */
+    public final void openIndexAsync(OpenIndexRequest openIndexRequest, ActionListener<OpenIndexResponse> listener, Header... headers) {
+        performRequestAsyncAndParseEntity(openIndexRequest, Request::openIndex, OpenIndexResponse::fromXContent, listener, emptySet(),
+                headers);
+    }
```

Comment:
we don't need these two methods, they are already in `IndicesClient`. Can you remove them?

Reply:
facepalm